worker/src/main/java/gov/cms/ab2d/worker/processor/ContractProcessorImpl.java

```diff
@@ -140,25 +173,42 @@ public ContractProcessorImpl(ContractRepository contractRepository,
             log.info("Finished writing {} EOBs for contract {}",
                     jobProgressService.getStatus(job.getJobUuid()).getEobsProcessedCount(), contractNumber);
 
+            // Mark the job as finished for the aggregator (all file data has been written out)
+            JobHelper.workerFinishJob(efsMount + "/" + job.getJobUuid() + "/" + streamingDir);
 
-            // Close the last file and report it as a job output
-            helper.closeLastStream();
-
-            List<StreamOutput> dataOutputs = helper.getDataOutputs();
-            dataOutputs.stream().map(output -> createJobOutput(output, false)).forEach(jobOutputs::add);
+            // Wait for the aggregator to finish
+            while (!aggregatorFuture.isDone()) {
```

Comment:
Is there some sort of timeout here or failure? It is totally possible to get stuck here forever and lock up the worker as a whole.

Reply:
Well, lock up that job thread. It is being stopped if the job is cancelled. I'll try to think of a place to check to kill it if we see other issues or exceptions. Time out? After how many hours?

Reply:
I guess the only problem is if the aggregator itself hangs.

Reply:
Failure of the aggregator is fine. But is failure of the aggregator going to be handled gracefully here?